File: src/Entities/Medicine.Entities/Models/Therapy.cs
using Medicine.Entities.Enums;
using Medicine.Entities.Models.Base.Interfaces;
using Medicine.Entities.Models.Translated;

namespace Medicine.Entities.Models
{
    /// <summary>
    /// Therapy can be assigned to user , or crate manualy 
    /// Each therapy can contain many courses
    /// </summary>
    public class Therapy : TranslationEntity<TranslatedTherapy>
    {
        public int UserId { get; set; }
        public TherapyStatus Status { get; set; } = TherapyStatus.None;
        public TherapyType Type { get; set; } = TherapyType.None;

        public List<Course> Courses { get; set; } = new List<Course>();

    }
}

File: front/medicine/src/app/shared/models/entity/Therapy.ts
import { TherapyType } from "../../enums/TherapyType";
import { TherapyStatus } from "../../enums/TherapyStatus";
import { User } from "./User";
import { Course } from "./Course";
import { TranslatedTherapy } from "./translate/TranslatedTherapy";

export class Therapy {
  userId!: number;
  status!: TherapyStatus;
  type!: TherapyType;
  courses: Course[] = [];
  translations: TranslatedTherapy[] = [];
  user?: User;
  id!: number;
  createdAt?: Date;
}
